lib/plugins/wallet/cardano/api.js

```diff
@@ -0,0 +1,161 @@
+const BN = require('../../../bn')
+
+const unitScale = 6
+
+class CardanoAPI {
+  constructor(client, port, hostname = 'localhost') {
+    this.client = client
+    this.port = port
+    this.hostname = hostname
+  }
+
+  async fetch(endpoint, params, data, appendData = []) {
+    const method = data ? 'POST' : 'GET'
+
+    const url = `https://${this.hostname}:${this.port}/api/v1/${endpoint}`
+
+    return Promise.resolve(true)
+      .then(() => {
+        return this.client({
+          method,
+          url,
+          params,
+          data
+        })
+      })
+      .then(response => {
+        if (response.error) throw r.error
+
+        const responseData = Array.isArray(response.data.data)
+          ? [...response.data.data, ...appendData]
+          : response.data.data
```

Comment:
this seems suspicious that we don't append anything even if `appendData` might not be empty list